\hypertarget{index_initial_sec}{}\section{Initial Project\+: The Wright-\/\+Fischer Model}\label{index_initial_sec}
{\bfseries -\/e} output file {\itshape  (default\+: terminal)} {\bfseries -\/r} number repeats {\itshape  (default\+: 3)} {\bfseries -\/g} number generations {\itshape  (default\+: 10)} {\bfseries -\/n} number individuals {\itshape  (default\+: 100)} {\bfseries  no flag } alleles frequencies {\itshape  (default\+: 0.\+1 0.\+2 0.\+3 0.\+4)}

Initially we only make a simple model of population genetics using the Wright Fisher model\+: there is population of N individual, they mate randomly and \char`\"{}mix\char`\"{} their D\+NA to produce the next generation of N individual \hypertarget{index_lackBiology}{}\subsection{Lack of Biology}\label{index_lackBiology}
We only model genetic drift, no selection, mutation, migration and no change in population size. For this basic model we don\textquotesingle{}t need to take into consideration the sex of the individuals and we can even only take into consideration the N$\ast$2 alleles and not the fact that they are grouped by 2. \hypertarget{index_randomMatting}{}\subsection{Random Matting\+:}\label{index_randomMatting}
 \hypertarget{index_results}{}\subsection{Results \& Plots}\label{index_results}
  \hypertarget{index_mutation_sec}{}\section{Mutation Extension}\label{index_mutation_sec}
{\bfseries -\/m} mutation probability {\itshape  (default\+: 0)}

To introduce mutation (probability of mutating\+: m) we just need to substract to the number of each allele $ N_i $ the number of alleles which mutates $ N_i*m $ and add the number of the alleles which mutate into $ N_i $ which is the number of other alleles times m $ (N_{tot}-N_i)*m $ divided by the number of alleles in which they can mutate (note that you can\textquotesingle{}t mutate into you) so\+: $ N_i'= N_i - N_i*m +\frac{(N_{tot}-N_i)*m}{A-1} $ finally we divide by the number of total alleles $ N_{tot} $ to obtain frequencies\+: \[ f_i'= f_i - f_i*m +\frac{(1-f_i)*m}{A-1}= f_i(1-m-\frac{m}{A-1}) + \frac{m}{A-1} = f_i(1-\frac{A*m}{A-1}) + \frac{m}{A-1} \] \hypertarget{index_selection_sec}{}\section{Natural Selection Extension}\label{index_selection_sec}
{\bfseries -\/s} alleles fitnesses {\itshape  (default\+: 0 0 0 ...)}

To introduce natural selection we just need to multiply $ f_i $ by a $ s_i $ which can be between -\/1 and infinity (0 if no selection) and then in order to have a constant population size you divide by $ \sum f_k*(1+s_k)\ $ so\+: \[ f_i'= \frac{f_i*(1+s_i)}{\sum f_k*(1+s_k)} = \frac{f_i*(1+s_i)}{\sum f_k +\sum f_ks_k} = \frac{f_i*(1+s_i)}{1 +\sum f_ks_k} \] \hypertarget{index_botlleneck_sec}{}\section{Bottleneck Extension}\label{index_botlleneck_sec}
{\bfseries -\/b} bottleneck effect flag {\itshape  (default\+: no)}

The bottleneck effect is a sharp reduction of the population size due to envirenmental events (earthquakes, fires ...) or human activities (genocides...). It enables a huge effect of genetic drift (random selection). In this case it creates a natural disaster which randomly kills 90\% of the population. \hypertarget{index_anemia_sec}{}\section{Sickle Cell and Malaria Extension}\label{index_anemia_sec}
{\bfseries -\/a} sickle cell anemia flag {\itshape  (default\+: no)}

In this \char`\"{}main\char`\"{} extension we wanted to make a model of genetic population more complex than the basic model. We chose to study the population genetics of sickle cell disease while taking into account malaria\+:
\begin{DoxyItemize}
\item Sickle cell anemia is indeed often taken as an example of natural selection. This disease is an inherited red blood cell disorder which comes from an abnormal hemoglobin coded by the recessive S allele of hemoglobin gene. The homozygous SS genotype is deadly in approximatively 70\% of cases. The heterozygous NS genotype causes some complications but doesn\textquotesingle{}t normally result in death (we didn\textquotesingle{}t take into complications the low death rate of NS), but prevents the patient of dying from malaria because the parasites cannot reproduce in the red blood cells. This adaptative advantage explains why drepanocytosis (sickle cell disease) is very prevalent in regions touched by malaria.
\item We chose 3 countries\+: Cameroon, Gabon and the Republic of the Congo, to try to make a model of the popultaion genetics while taking into account the real statistics.
\item In order to make a better model we also computed the migration between these countries. Note that we didn\textquotesingle{}t take into account the population growth, mutations, other migrations with other countries, non random matting. The statistics (population size, malaria pourcentage, sickle cell disease pourcentage, population migration...) were taken from {\bfseries W\+HO}  -\/$>$ Using basic probability we found the new probabiliy of getting a N and a S\+: $ f(N)= (1-m)*p^{2} + p*q $ and $ f(S)= 0.3*q^{2} + p*q $
\end{DoxyItemize}

